src/scaler/spreadscaler.rs

```diff
@@ -0,0 +1,316 @@
+use anyhow::Result;
+use async_trait::async_trait;
+use std::{
+    cmp::{self, Ordering},
+    collections::HashSet,
+};
+
+use crate::{
+    commands::{Command, StartActor, StopActor},
+    events::Event,
+    model::{Spread, SpreadScalerProperty},
+    scaler::Scaler,
+    storage::{Actor, Host, ReadStore},
+};
+
+/// Config for an ActorSpreadScaler
+struct ActorSpreadConfig {
+    actor_reference: String,
+    lattice_id: String,
+    spread_config: SpreadScalerProperty,
+}
+
+/// The SimpleScaler ensures that a certain number of replicas are running
+/// for a certain public key.
+///
+/// This is primarily to demonstrate the functionality and ergonomics of the
+/// [Scaler](crate::scaler::Scaler) trait and doesn't make any guarantees
+/// about spreading replicas evenly
+struct ActorSpreadScaler<S: ReadStore + Send + Sync> {
+    pub config: ActorSpreadConfig,
+    store: S,
+}
+
+#[async_trait]
+impl<S: ReadStore + Send + Sync> Scaler for ActorSpreadScaler<S> {
+    type Config = ActorSpreadConfig;
+
+    fn update_config(&mut self, config: Self::Config) -> Result<bool> {
```

Comment:
Was just thinking about this: Update config should probably trigger a reconcile

Reply:
Hm, true. Maybe this returns a HashSet of commands too